lightning/src/chain/channelmonitor.rs

```diff
@@ -1639,24 +1738,24 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 						if let Some(ref outpoints) = self.counterparty_claimable_outpoints.get($txid) {
 							for &(ref htlc, ref source_option) in outpoints.iter() {
 								if let &Some(ref source) = source_option {
-									log_info!(logger, "Failing HTLC with payment_hash {} from {} counterparty commitment tx due to broadcast of revoked counterparty commitment transaction, waiting for confirmation (at height {})", log_bytes!(htlc.payment_hash.0), $commitment_tx, height + ANTI_REORG_DELAY - 1);
-									match self.onchain_events_waiting_threshold_conf.entry(height + ANTI_REORG_DELAY - 1) {
-										hash_map::Entry::Occupied(mut entry) => {
-											let e = entry.get_mut();
-											e.retain(|ref event| {
-												match **event {
-													OnchainEvent::HTLCUpdate { ref htlc_update } => {
-														return htlc_update.0 != **source
-													},
-													_ => true
-												}
-											});
-											e.push(OnchainEvent::HTLCUpdate { htlc_update: ((**source).clone(), htlc.payment_hash.clone())});
+									self.onchain_events_waiting_threshold_conf.retain(|ref entry| {
```

Comment:
Would you like me to add this to the corresponding methods in `ChainMonitor` and `ChannelMonitor`? I'm a wondering if we should move this to some module-level documentation and reference it rather than repeating the comment. Though note that the code is across two different modules currently.

Reply:
Yea, I was thinking after writing this that maybe a parallel trait to `chain::Listen` may make sense.

Reply:
Any objection to expanding `chain::Listen` to include the three new methods? Otherwise, I can ponder a good name for a new trait.

Reply:
Hmm, I guess we could, but it feels strange to me. Really you should only ever use one or the other, never both. Sure, different traits isn't really an enforcement of that, but it at least exposes it somewhat.

Reply:
Good point. `chain::Update` is the first thing that comes to mind but will let the subconscious work on it while making the change.

Reply:
Will make this change in a follow-up PR as discussed offline.